helm/falco/values.yaml

```diff
@@ -24,6 +24,17 @@ falco:
         image:
           registry: gsoci.azurecr.io
           repository: giantswarm/falco-driver-loader
+  customRules:
+    gs-root-ca-key-access.yaml: |-
+      - list: root_ca_key_path
+        items:
+        - "/etc/kubernetes/pki/ca.key"
+      - rule: File Related Operation on Root CA Key Path
+        desc: Detect any file operation on the path to root CA key
+        condition: (fs.path.name pmatch (root_ca_key_path) or fs.path.source pmatch (root_ca_key_path) or fs.path.target pmatch (root_ca_key_path))
+        output: "Some File Related Operation on root CA key (evt.type=%evt.type path=%fs.path.name source=%fs.path.source target=%fs.path.target user.name=%user.name proc.cmdline=%proc.cmdline proc.pcmdline=%proc.pcmdline container.id=%container.id container.name=%container.name image=%container.image.repository)"
```

Comment:
Are the Falco fields specific enough to make this message more of a story for the responder? For a rough example, something like:
```
Root CA key file was "< opened | written | deleted | whatever >" by the process "< process name >" from the container "< namespace / container name >" on the node "< node >"
```
The image or container names might be empty in some cases, but just for a general idea. Is that possible?

Reply:
I think all details available for that kind of operation are already present in the message, but I suppose they can be rearranged to make the message more user friendly.
Btw. any access to the root CA key file (with the exception of access from internal Kubernetes components like kubelet, controller-manager, etc.) is considered bad.

Reply:
It would be awesome to already have that kind of message in that case, so if/when we get false positives in the beginning, responders can work with the alerts more easily

Reply:
Is this any better? 
```
Root CA key file '/etc/kubernetes/pki/ca.key' was accessed via operation 'openat' (source path '<NA>', target path '<NA>') triggered by the user 'root' running bash process 'less ca.key' in container 'host' (ID=host) built from image '<NA>'. Additional details (if available): container_id=host container_image=<NA> container_image_tag=<NA> container_name=host k8s_ns=<NA> k8s_pod_name=<NA>
```

Btw. Falco appends the last few properties automatically.

I know it's probably still not ideal, but I am not sure I can do much better than that because different file operations originating in different places have different details available.

Regarding the false alarms - I was thinking to print a message to stdout first and try to identify legitimate uses before I go on and implement the alerting. And if it's desitrable to implement the alerting as soon as possible, I would still prefer to run it and just print messages to stdout for some time to at least see how many false alarms we can expect.

Reply:
> Is this any better?
That is beautiful actually :heart_eyes:

Reply:
OK, thanks. I will update the rule.

Reply:
Done